components/scream/src/share/field/field_property_checks/field_monotonicity_check.hpp

```diff
@@ -23,10 +23,10 @@ class FieldMonotonicityCheck: public FieldPropertyCheck<RealType> {
   // Overrides.
 
   // The name of the field check
-  std::string name () const { return "Monotonicity Field Check"; }
+  std::string name () const override { return "Monotonicity Field Check"; }
 
   bool check(const Field<const_RT>& field) const override {
-    auto view = field.get_view();
+    auto view = field.get_flattened_view();
```

Comment:
A general comment for all of the field checks.  Now that `get_flattened_view` will throw an error if a field is actually a subfield we will probably want to rethink this approach.  At this point it would not be possible to field check any of the tracers.

Since we are not performing the field checking by default we don't need to fix this yet.  But will want to address this sooner than later.

Reply:
Yes, that's true. I think it won't be long till we actually need that fix. If you think it's fine, I can add it to this PR.

Reply:
I'm happy to have it added here if it's not too much work.